runtime/src/utxo.rs

```diff
@@ -221,38 +236,25 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
-	/// Redistribute combined reward value evenly among chain authorities
-	fn disperse_reward(authorities: &[H256]) {
-		let reward = <RewardTotal>::take();
-		let share_value: Value = reward
-			.checked_div(authorities.len() as Value)
-			.ok_or("No authorities")
-			.unwrap();
-		if share_value == 0 { return }
-
-		let remainder = reward
-			.checked_sub(share_value * authorities.len() as Value)
-			.ok_or("Sub underflow")
-			.unwrap();
-		<RewardTotal>::put(remainder as Value);
-
-		for authority in authorities {
-			let utxo = TransactionOutput {
-				value: share_value,
-				pubkey: *authority,
-			};
-
-			let hash = BlakeTwo256::hash_of(&(&utxo,
-						<system::Module<T>>::block_number().saturated_into::<u64>()));
-
-			if !<UtxoStore>::contains_key(hash) {
-				<UtxoStore>::insert(hash, utxo);
-				sp_runtime::print("transaction reward sent to");
-				sp_runtime::print(hash.as_fixed_bytes() as &[u8]);
-			} else {
-				sp_runtime::print("transaction reward wasted due to hash collision");
-			}
-		}
+	/// Redistribute combined reward value to block Author
+	fn disperse_reward(author: &Public) {
+		let reward = RewardTotal::take() + T::Issuance::issuance(system::Module::<T>::block_number());
+
+		let utxo = TransactionOutput {
+			value: reward,
+			// TODO Is this the correct way to convert public key into H256?
+			// I cribbed this logic from chain_spec.rs
+			pubkey: H256::from_slice(author.as_slice()),
+		};
+
+		let hash = BlakeTwo256::hash_of(&(&utxo,
+					<system::Module<T>>::block_number().saturated_into::<u64>()));
+
+		//TODO consider removing these `print`s
```

Comment:
yeah it was in the previous impl, I thought about removing it as well, but it works well for the demo (otherwise hard to get the utxo hash to verify), teachers workshoppers how to log to console, and is generally pretty helpful. lets just keep this in?

Reply:
Leaving them in for the time being just so we can be sure that `on_finalize` is actually called. Will take them out shortly.